spa_ui/self_service/client/app/states/marketplace/list/list.state.js

```diff
@@ -45,20 +46,37 @@
       $state.go('marketplace.details', {serviceTemplateId: template});
     }
 
+    var categoryNames = [];
+
+    angular.forEach(vm.serviceTemplates, function(item) {
+      if (typeof item.service_template_catalog !== 'undefined' 
```

Comment:
Angular provides a helper `isUndefined()` (and counterpart `isDefined()`) that can help with the readability of `undefined` checks.

ex. `angular.isDefined(item.service_template_catalog)`